rules/internal/utils/type_name.go

```diff
@@ -17,23 +17,17 @@ package utils
 import (
 	"strings"
 
-	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
 	"github.com/jhump/protoreflect/desc"
 )
 
-type hasGetType interface {
-	GetType() dpb.FieldDescriptorProto_Type
-}
-
-// GetScalarTypeName returns the name of the type of the field, as a string.
-func GetScalarTypeName(t hasGetType) string {
-	return strings.ToLower(t.GetType().String()[len("TYPE_"):])
-}
-
-// GetMessageTypeName returns the name of the type of the field, as a string.
-func GetMessageTypeName(f *desc.FieldDescriptor) string {
-	if messageType := f.GetMessageType(); messageType != nil {
-		return messageType.GetFullyQualifiedName()
+// GetTypeName returns the name of the type of the field, as a string,
+// regardless
+func GetTypeName(f *desc.FieldDescriptor) string {
```

Comment:
Add a support for map type too.

https://godoc.org/github.com/jhump/protoreflect/desc#FieldDescriptor.GetMapKeyType
https://godoc.org/github.com/jhump/protoreflect/desc#FieldDescriptor.GetMapValueType

Reply:
Reasonable, but can I do it in a follow-up?

Reply:
Sure. Can you add a TODO?

Reply:
```suggestion
//
// TODO: Add support for map types.
func GetTypeName(f *desc.FieldDescriptor) string {
```